src/main/java/appeng/me/storage/AbstractCellInventory.java

```diff
@@ -51,46 +51,37 @@
 	protected static final String ITEM_PRE_FORMATTED_SLOT = "PF#";
 	protected static final String ITEM_PRE_FORMATTED_NAME = "PN";
 	protected static final String ITEM_PRE_FORMATTED_FUZZY = "FP";
-	private static String[] itemSlots;
-	private static String[] itemSlotCount;
+	private static final String[] ITEM_SLOT_KEYS = new String[MAX_ITEM_TYPES];
+	private static final String[] ITEM_SLOT_COUNT_KEYS = new String[MAX_ITEM_TYPES];
 	private final NBTTagCompound tagCompound;
 	protected final ISaveProvider container;
-	private int maxItemTypes = 63;
+	private int maxItemTypes = MAX_ITEM_TYPES;
 	private short storedItems = 0;
 	private int storedItemCount = 0;
 	protected IItemList<T> cellItems;
-	protected ItemStack i;
-	protected IStorageCell<T> cellType;
+	protected final ItemStack i;
+	protected final IStorageCell<T> cellType;
 	protected final int itemsPerByte;
+	private boolean isPersisted = true;
 
-	protected AbstractCellInventory( final NBTTagCompound data, final ISaveProvider container, final int itemsPerByte )
+	static
 	{
-		this.tagCompound = data;
-		this.container = container;
-		this.itemsPerByte = itemsPerByte;
+		for( int x = 0; x < MAX_ITEM_TYPES; x++ )
+		{
+			ITEM_SLOT_KEYS[x] = ITEM_SLOT + x;
+			ITEM_SLOT_COUNT_KEYS[x] = ITEM_SLOT_COUNT + x;
+		}
 	}
 
 	protected AbstractCellInventory( final ItemStack o, final ISaveProvider container, final int itemsPerByte ) throws AppEngException
 	{
 		this.itemsPerByte = itemsPerByte;
-		if( itemSlots == null )
-		{
-			itemSlots = new String[this.maxItemTypes];
-			itemSlotCount = new String[this.maxItemTypes];
-
-			for( int x = 0; x < this.maxItemTypes; x++ )
-			{
-				itemSlots[x] = ITEM_SLOT + x;
-				itemSlotCount[x] = ITEM_SLOT_COUNT + x;
-			}
-		}
 
 		if( o == null )
 		{
 			throw new AppEngException( "ItemStack was used as a cell, but was not a cell!" );
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "ItemStack was used as a cell, but was not a cell!" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)